utils/restored-receipts-verifier/src/main.rs

```diff
@@ -0,0 +1,106 @@
+
+    let shard_id: u64 = 0;
+    let home_dir = matches.value_of("home").map(|dir| Path::new(dir)).unwrap();
+    let near_config = load_config(&home_dir);
+    let store = create_store(&get_store_path(&home_dir));
+    let mut chain_store = ChainStore::new(store.clone(), near_config.genesis.config.genesis_height);
+    let runtime = NightshadeRuntime::new(
+        &home_dir,
+        store,
+        &near_config.genesis,
+        near_config.client_config.tracked_accounts.clone(),
+        near_config.client_config.tracked_shards.clone(),
+    );
+
+    let mut receipts_missing: Vec<Receipt> = vec![];
+    let height_first: u64 = 34691244; // First height for which lost receipts were found
+    let height_last: u64 = 35524259; // Height for which apply_chunks was already fixed
+
+    for height in height_first..height_last {
+        let block_hash_result = chain_store.get_block_hash_by_height(height);
+        if block_hash_result.is_err() {
+            println!("{} does not exist, skip", height);
+            continue;
+        }
+        let block_hash = block_hash_result.unwrap();
+
+        let block = chain_store.get_block(&block_hash).unwrap().clone();
+        if block.chunks()[shard_id as usize].height_included() == height {
+            println!("{} included, skip", height);
+            continue;
+        }
+
+        let chunk_extra =
+            chain_store.get_chunk_extra(block.header().prev_hash(), shard_id).unwrap().clone();
+        let apply_result = runtime
+            .apply_transactions(
+                shard_id,
+                chunk_extra.state_root(),
+                block.header().height(),
+                block.header().raw_timestamp(),
+                block.header().prev_hash(),
+                &block.hash(),
+                &[],
+                &[],
+                chunk_extra.validator_proposals(),
+                block.header().gas_price(),
+                chunk_extra.gas_limit(),
+                &block.header().challenges_result(),
+                *block.header().random_value(),
+                false,
+            )
+            .unwrap();
+
+        let receipts_missing_after_apply: Vec<Receipt> =
+            apply_result.receipt_result.values().cloned().into_iter().flatten().collect();
+        receipts_missing.extend(receipts_missing_after_apply.into_iter());
+        println!("{} applied", height);
+    }
+
+    // Check that receipts from repo were actually generated
+    let receipt_result_in_repo_json =
+        include_str!("../../../neard/res/fix_apply_chunks_receipts.json");
+    let receipt_result_in_repo = serde_json::from_str::<ReceiptResult>(receipt_result_in_repo_json)
+        .expect("File with receipts restored after apply_chunks fix have to be correct");
+    let receipts_in_repo = receipt_result_in_repo.get(&shard_id).unwrap();
+    let receipt_hashes_in_repo =
+        HashSet::<_>::from_iter(receipts_in_repo.into_iter().map(|receipt| receipt.get_hash()));
```

Comment:
Nit: this bit of code is a bit hard to follow. The smell here is that we have a whole bunch of variables here, and they all start with the same prefix. So, to a fast-reading human mind, all this variables look essentially the same, and you need to concentrate to actaully read the names fully and to understand what goes where. 

A nice trick to make the data flow here more obvious is to use a block expression: 

```rust
    let receipt_hashes_missing: HashSet<CryptoHash> =
        receipts_missing.into_iter().map(|receipt| receipt.get_hash()).collect();

    let receipt_hashes_in_repo: HashSet<CryptoHash> = {
        // Check that receipts from repo were actually generated
        let receipt_result_in_repo_json =
            include_str!("../../../neard/res/fix_apply_chunks_receipts.json");
        let receipt_result_in_repo =
            serde_json::from_str::<ReceiptResult>(receipt_result_in_repo_json)
                .expect("File with receipts restored after apply_chunks fix have to be correct");
        let receipts_in_repo = receipt_result_in_repo.get(&shard_id).unwrap();
        receipts_in_repo.into_iter().map(|receipt| receipt.get_hash()).collect()
    };
```

There's still a bunch of names here, and they are still too repetitive, but now it is clear to the reader that only the two top-level ones are worth paying attention to.